bot.js

```diff
@@ -209,16 +214,64 @@ function memberUpdateHandler(oldMember, newMember) {
     }
   }
 }
+// Also run once on startup of script
+// Will post if all conditions are met (time is correct, post is not stickied, post have not been posted)
+const redditInterval = async () => {
+  const shouldPost = isRedditPostingTime();
+  if (!shouldPost) return;
+  try {
+    const res = await Phin({ url: REDDIT_URL, parse: "json" });
+    const postList = res.body.data.children;
+    let validPost = null;
+    let startAt = 0;
+    while (!validPost) {
+      const [postLink, position] = getNthNonStickiedPost(postList, startAt);
+      if (await havePostedAlready(postLink)) {
+        startAt = position + 1;
+        continue;
+      } else {
+        validPost = postLink;
+      }
+    }
+    const channel = await client.channels.get(REDDIT_POSTING_CHANNEL_ID);
+    channel.send(validPost);
+  } catch(error) {
+    console.error(error);
+  } 
+}
+
+
+
+
+
 
 const limitedMessageHandler = RateLimiter(COMMAND_COOLDOWN, msgHandler);
 client.on("message", limitedMessageHandler);
 
 client.on("guildMemberUpdate", memberUpdateHandler);
 
-// Also run once on startup of script
+
 // Every hour check if the current time is within the ranges
-setInterval(() => {
-  const shouldPost = isRedditPostingTime();
-}, 60 * 60 * 1000);
+setInterval(redditInterval, 60 * 60 * 1000);
+
+// Takes a post, finds out if we've already posted it in the relevant channel, returns boolean.
+async function havePostedAlready(postLink) {
```

Comment:
Let's move both of these functions into helpers.js. You'll need to import them at the top of the file in order to use them though.